src/components/scroll-monitor.tsx

```diff
@@ -2,48 +2,58 @@ import * as React from 'react'
 import { findDOMNode } from 'react-dom'
 
 interface Props {
+  page: number
   onPage: (number: number) => void
 }
 
+const noop = () => {}
+
 class ScrollMonitor extends React.Component<Props, null> {
-  page = 0
-  threshold = 0
-  teardown: () => void
+  teardown: () => void = noop
```

Comment:
Can you help me understand what this is doing?  I'm a little new to Typescript. I think this is saying the function should return a void type, like null or undefined? Is that too restrictive to require other devs to return null or undefined, when really we're just saying that this component doesn't care about the output?

Reply:
@dbernheisel There is no void type in javascript.  A function with no return will return undefined.  This just says, I never care about the return value, and tells the compiler to never let you use the return value for anything.